lib/web/apiserver.go

```diff
@@ -2148,15 +2149,26 @@ func clientMetaFromReq(r *http.Request) *auth.ForwardedClientMetadata {
 	}
 }
 
-// deleteWebSession is called to sign out user
+// deleteWebSession is called to sign out user from web, app and SAML IdP session.
 //
 // DELETE /v1/webapi/sessions/:sid
 //
 // Response:
 //
 // {"message": "ok"}
 func (h *Handler) deleteWebSession(w http.ResponseWriter, r *http.Request, _ httprouter.Params, ctx *SessionContext) (interface{}, error) {
-	err := h.logout(r.Context(), w, ctx)
+	// TODO(sshah): we can skip this step below once we have a mechanism to update websession
+	// with SAML details.
+	samlSessionCookie, err := r.Cookie(samlidp.SAMLSessionCookieName)
+	if err != nil {
+		h.log.Debug("SAML IdP session not found for deletion for user: %s", ctx.GetUser())
+	} else if samlSessionCookie != nil && samlSessionCookie.Value != "" {
+		if ctx.cfg.Session.GetSAMLSession() == nil {
```

Comment:
Why does this need to happen here?

Would it be possible for a SAML session to already exist, but point at a different cookie?

Reply:
> Why does this need to happen here?

Only because this value is never set until now. This PR sets it during logout, but that too happens only if SAML session cookie is present.

> Would it be possible for a SAML session to already exist, but point at a different cookie?

In the same browser? No. Not unless user tries to spoof session cookie value.
If that happens then the SAML IdP detects session ID mismatch and fails the login process.

Reply:
Sounds good. Could you add a comment with some of that background?

Reply:
Bump for this one? I'll leave it approve so I don't block you further.

Reply:
https://github.com/gravitational/teleport/pull/37619/commits/dc53d1730a81b43b7a7cab402e5a1c4fc7678aab